unified-runtime/scripts/core/PROG.rst

```diff
@@ -10,6 +10,26 @@
  Programming Guide
 ========================
 
+Adapters
+========
+
+The top level abstraction in the unified runtime is the adapter handle. It owns
+the platform handles as well as any backend specific global state, for instance
+if a backend has global setup/teardown type entry-points these will typically
+be tied to the adapter handle's constructor and destructor.
+
+Adapter handles are reference counted via ${x}AdapterRetain and
+${x}AdapterRelease. Once their reference count reaches zero any objects
+associated with the adapter in question should be considered invalid. Calling
+${x}AdapterGet after any adapter handle's reference count has reached zero will
```

Comment:
Is it still UB if `urLoaderTearDown` followed by `urLoaderInit` is called?

Reply:
I wonder if our test suites do that anywhere.

Reply:
https://github.com/intel/llvm/issues/17861